piptools/scripts/compile.py

```diff
@@ -159,6 +159,12 @@ def _get_default_option(option_name: str) -> Any:
         )
     ),
 )
+@click.option(
+    "--constraints-compatible",
```

Comment:
I'm not sure if this name is obvious enough. I'd call it `--strip-extras` maybe...